src/llama-context.cpp

```diff
@@ -2274,6 +2278,11 @@ llama_context * llama_init_from_model(
         params.flash_attn = false;
     }
 
+    if (params.flash_attn && model->arch == LLM_ARCH_DEEPSEEK2) {
+        LLAMA_LOG_WARN("%s: flash_attn is not compatible with Deepseek2 - forcing off\n", __func__);
+        params.flash_attn = false;
+    }
+
```

Comment:
There is a way to enable the FA path to be compatible with MLA. I will fix this in a follow-up PR https://github.com/ggml-org/llama.cpp/tree/gg/mla. For now, keep it like this.